### PR TITLE
Add support for docker credentials

### PR DESCRIPTION
**Pull request by jthiltges**
- Include username/password when requesting access token
- Pass docker credentials to singularity

### CODE REVIEW

File: cvmfs-singularity-sync [hunk 1]
```diff
@@ -338,7 +355,18 @@ def parse_image(image):
 
     return registry, namespace, repo_name, repo_tag
 
-def publish_image(image, singularity_rootfs, registry, doauth, socket=None):
+def get_manifest(hub, namespace, repo_name, repo_tag, manifest_cache):
+    metadata = hub.manifest(namespace, repo_name, repo_tag, head=True)
+    digest = metadata.headers['docker-content-digest']
+
+    if digest in manifest_cache:
+        return manifest_cache['digest']
```
Comment (matyasselmeci) [inline cvmfs-singularity-sync:363]:
```suggestion
        return manifest_cache[digest]
```

Comment (jthiltges) [reply cvmfs-singularity-sync:363]: Yikes. Good catch! Thanks.

**Review by matyasselmeci — APPROVED** [review]
Other than that typo I think it looks fine, though if you're going to store docker credentials in the DB, you might want to make sure the file is not world-readable.

**Review by djw8605 — APPROVED** [review]
Neat!  Approved

**Pull request merged**